spec/velveteen/worker_spec.rb

```diff
@@ -64,6 +64,61 @@
       _, properties, _ = queue.pop
       expect(properties[:headers]).to eq("foo" => "bar")
     end
+
+    describe "payload validation" do
+      it "raises an error when payload does not match schema" do
+        message = Velveteen::Message.new(
+          data: {"foo" => "bar"},
+          headers: {}
+        )
+        invalid_message = Velveteen::Message.new(
+          data: {"coffee" => "espresso"},
+          headers: {}
+        )
+        publish_routing_key = "worker.test.tea.ordered"
+        worker = TestValidatesPayloadWorker.new(
+          message: message,
+          publish_message: invalid_message,
+          publish_routing_key: publish_routing_key
+        )
+        queue = Velveteen::Config.channel.queue
+        queue.bind(
+          Velveteen::Config.exchange,
+          routing_key: publish_routing_key
+        )
+
+        expect {
+          worker.perform
+        }.to raise_error(Velveteen::InvalidMessage, /tea/)
+      end
+
+      it "publishes when payload matches routing key schema" do
+        message = Velveteen::Message.new(
+          data: {"foo" => "bar"},
+          headers: {}
+        )
+        message_to_publish = Velveteen::Message.new(
```

Comment:
What do you think about not building a message object for this and renaming it to `payload`?